src/structure.js

```diff
@@ -277,15 +280,14 @@ Structure.prototype.reference = function (path) {
      * @returns {Void}
      */
     destroy: function () {
-      removeAllListenersBut(self, pathId, referenceListeners);
-      referenceListeners = void 0;
-      cursor = void 0;
+      removeAllListenersBut(listenerNs);
+      cursor = undefined;
 
       this._dead = true;
-      this.observe = void 0;
-      this.unobserveAll = void 0;
-      this.cursor = void 0;
-      this.destroy = void 0;
+      this.observe = undefined;
```

Comment:
Will clean up this PR or submit a better one if I can't figure out the git commands to fix it. The key sections are: listListenerMatching, getListenerNs, and the reference constructor